SIGSEGV in github.com/nats-io/nats-server/v2/server.(*stream).startClusterSubs during node restart

### Observed behavior

The behavior is observed with 3 nodes NATS cluster deployed via helm with JWT authentication.
We have few queues all replicated 3 times. From time to time during node restart we're observing SIGSEGV errors as below:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x8e7c14]

goroutine 72 [running]:
github.com/nats-io/nats-server/v2/server.(*stream).startClusterSubs(...)
github.com/nats-io/nats-server/v2/server/stream.go:1025
github.com/nats-io/nats-server/v2/server.(*jetStream).processClusterUpdateStream(0xc0000238c0, 0xc000260a88, 0xc0002df8c0, 0xc00032c990)
github.com/nats-io/nats-server/v2/server/jetstream_cluster.go:3664 +0x414
github.com/nats-io/nats-server/v2/server.(*jetStream).processUpdateStreamAssignment(0xc0000238c0, 0xc00032c990)
github.com/nats-io/nats-server/v2/server/jetstream_cluster.go:3590 +0x413
github.com/nats-io/nats-server/v2/server.(*jetStream).applyMetaEntries(0xc0000238c0, {0xc000428320, 0x2, 0x0?}, 0x0)
github.com/nats-io/nats-server/v2/server/jetstream_cluster.go:1973 +0x9cc
github.com/nats-io/nats-server/v2/server.(*jetStream).monitorCluster(0xc0000238c0)
github.com/nats-io/nats-server/v2/server/jetstream_cluster.go:1284 +0xcf0
github.com/nats-io/nats-server/v2/server.(*Server).startGoRoutine.func1()
github.com/nats-io/nats-server/v2/server/server.go:3891 +0x32
created by github.com/nats-io/nats-server/v2/server.(*Server).startGoRoutine in goroutine 1
github.com/nats-io/nats-server/v2/server/server.go:3889 +0x128
```

Which results in crash loop.


### Expected behavior

No crash during restarts of a single node.

### Server and client version

Tried both with `2.10.24-alpine` and `2.11.8-alpine`. Helm chart is `v1.3.13`

### Host environment

Ubuntu 22.04/Kubernetes 1.31.10

### Steps to reproduce

Not sure how to reproduce it exactly. It happens randomly.

--- a/server/jetstream_cluster.go
+++ b/server/jetstream_cluster.go
@@ -4933,6 +4933,9 @@ func (js *jetStream) processClusterUpdateStream(acc *Account, osa, sa *streamAss
 
 		if !alreadyRunning && numReplicas > 1 {
 			if needsNode {
+				// Must run before startClusterSubs reads mset.sa.Sync.
+				mset.setStreamAssignment(sa)
+
 				// Since we are scaling up we want to make sure our sync subject
 				// is registered before we start our raft node.
 				mset.mu.Lock()

--- a/server/jetstream_cluster_1_test.go
+++ b/server/jetstream_cluster_1_test.go
@@ -6785,6 +6785,72 @@ func TestJetStreamClusterMetaRecoveryAddAndUpdateStream(t *testing.T) {
 	require_Len(t, len(sa.Config.Subjects), 1)
 }
 
+// https://github.com/nats-io/nats-server/issues/7229
+func TestJetStreamClusterProcessClusterUpdateStreamNilStreamAssignment(t *testing.T) {
+	c := createJetStreamClusterExplicit(t, "R3S", 3)
+	defer c.shutdown()
+
+	nc, js := jsClientConnect(t, c.randomServer())
+	defer nc.Close()
+
+	_, err := js.AddStream(&nats.StreamConfig{Name: "TEST", Subjects: []string{"foo"}, Replicas: 3})
+	require_NoError(t, err)
+
+	s := c.randomServer()
+	acc, err := s.lookupAccount(globalAccountName)
+	require_NoError(t, err)
+	mset, err := acc.lookupStream("TEST")
+	require_NoError(t, err)
+
+	// Capture osa and build a sa that reflects the post-restart meta-replay
+	// conditions: a scale-up path where both the old and new raftGroup nodes
+	// are nil on this server (alreadyRunning=false, needsNode=true).
+	sjs := s.getJetStream()
+	sjs.mu.Lock()
+	realOsa := sjs.streamAssignment(globalAccountName, "TEST")
+	if realOsa == nil {
+		sjs.mu.Unlock()
+		t.Fatal("stream assignment not found")
+	}
+
+	// Intentionally omit Group.node so that in processClusterUpdateStream
+	// alreadyRunning (osa.Group.node != nil) is false and needsNode
+	// (sa.Group.node == nil) is true, taking the scale-up branch.
+	cloneGroup := func() *raftGroup {
+		return &raftGroup{
+			Name:    realOsa.Group.Name,
+			Peers:   append([]string{}, realOsa.Group.Peers...),
+			Storage: realOsa.Group.Storage,
+		}
+	}
+	osa := *realOsa
+	osa.Group = cloneGroup()
+	sa := osa
+	cfg := *realOsa.Config
+	cfg.MaxMsgs = 1_000
+	sa.Config = &cfg
+	sa.Group = cloneGroup()
+	sjs.mu.Unlock()
+
+	// Simulate the state right after recoverStream() before meta catchup:
+	// no stream assignment, no raft node, no cluster subs registered yet.
+	mset.mu.Lock()
+	mset.sa = nil
+	mset.node = nil
+	if mset.syncSub != nil {
+		mset.srv.sysUnsubscribe(mset.syncSub)
+		mset.syncSub = nil
+	}
+	mset.mu.Unlock()
+
+	sjs.processClusterUpdateStream(acc, &osa, &sa)
+
+	mset.mu.RLock()
+	got := mset.sa
+	mset.mu.RUnlock()
+	require_NotNil(t, got)
+}
+
 // Make sure if we received acks that are out of bounds, meaning past our
 // last sequence or before our first that they are ignored and errored if applicable.
 func TestJetStreamClusterConsumerAckOutOfBounds(t *testing.T) {

